src/Symfony/Component/Cache/Adapter/RedisAdapter.php

```diff
@@ -144,14 +147,16 @@ protected function doHave($id)
      */
     protected function doClear($namespace)
     {
+        // As documented in Redis documentation (http://redis.io/commands/keys) using KEYS
+        // can hang your server when it is executed against large databases (millions of items).
+        // Whenever you hit this scale, it is advised to deploy one Redis database per cache pool
+        // instead of using namespaces, so that FLUSHDB is used instead.
+        $lua = "local keys=redis.call('KEYS',ARGV[1]..'*') for i=1,#keys,5000 do redis.call('DEL',unpack(keys,i,math.min(i+4999,#keys))) end";
```

Comment:
Well, this other PR is changing the code anyway, so this does not forbid keeping this inside the ``else`` clause.